Source/WebCore/platform/mediastream/RealtimeMediaSource.cpp

```diff
@@ -426,12 +429,22 @@ void RealtimeMediaSource::end(Observer* callingObserver)
     m_isEnded = true;
     didEnd();
 
+    if (m_registerOwnerCallback)
+        m_registerOwnerCallback(*this, false);
+
     forEachObserver([&callingObserver](auto& observer) {
         if (&observer != callingObserver)
             observer.sourceStopped();
     });
 }
 
+void RealtimeMediaSource::registerOwnerCallback(OwnerCallback&& callback)
+{
+    ASSERT(isMainThread());
+    if (!m_registerOwnerCallback)
+        m_registerOwnerCallback = WTFMove(callback);
```

Comment:
Should this do something like `ASSERT(!m_registerOwnerCallback || m_registerOwnerCallback == callback)`?

Reply:
Will add the ASSERT, registration of a source should only happen once.